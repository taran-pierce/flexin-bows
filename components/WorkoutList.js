import Link from 'next/link'

const WorkoutList = (props) => {
  return (
    <div>
      <ul>
        {props.links.map( (link) => (
          <li>
            <Link href={link.href}>
              <a>{link.name}</a>
            </Link>
          </li>
        ))}
      </ul>
    </div>
  )
}

export default WorkoutList