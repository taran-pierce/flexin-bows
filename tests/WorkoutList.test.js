import React from 'react';
import WorkoutList from '../components/WorkoutList';
import renderer from 'react-test-renderer';

test('Does WorkoutList render correctly?', () => {
  const links = [
    {
      name: 'Better Body',
      href: '/better-body'
    },
    {
      name: 'Advanced Conditioning',
      href: '/advanced-conditioning'
    },
    {
      name: 'Body Building',
      href: '/body-building'
    },
    {
      name: 'Strength Training',
      href: '/strength-training'
    },
    {
      name: 'Aerobic',
      href: '/aerobic'
    },
    {
      name: 'Upper Lower Body',
      href: '/upper-lower-body'
    },
    {
      name: 'Anaerobic',
      href: '/anaerobic'
    }
  ]
  
  const component = renderer.create(
    <WorkoutList links={links} />,
  );
  
  // verify component compiles correctly
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
