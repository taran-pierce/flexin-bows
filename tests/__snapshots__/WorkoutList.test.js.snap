// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does WorkoutList render correctly? 1`] = `
<div>
  <ul>
    <li>
      <a
        href="/better-body"
        onClick={[Function]}
      >
        Better Body
      </a>
    </li>
    <li>
      <a
        href="/advanced-conditioning"
        onClick={[Function]}
      >
        Advanced Conditioning
      </a>
    </li>
    <li>
      <a
        href="/body-building"
        onClick={[Function]}
      >
        Body Building
      </a>
    </li>
    <li>
      <a
        href="/strength-training"
        onClick={[Function]}
      >
        Strength Training
      </a>
    </li>
    <li>
      <a
        href="/aerobic"
        onClick={[Function]}
      >
        Aerobic
      </a>
    </li>
    <li>
      <a
        href="/upper-lower-body"
        onClick={[Function]}
      >
        Upper Lower Body
      </a>
    </li>
    <li>
      <a
        href="/anaerobic"
        onClick={[Function]}
      >
        Anaerobic
      </a>
    </li>
  </ul>
</div>
`;
